tests/test_lookml.py

```diff
@@ -1123,13 +1202,88 @@ def test_lookml_actual(
                             "type": "count_distinct",
                         },
                     ],
-                }
+                },
+                {
+                    "dimensions": [
+                        {
+                            "hidden": "yes",
+                            "name": "document_id",
+                            "primary_key": "yes",
```

Comment:
`document_id` is not a primary key for this table, right? e.g.
```
SELECT document_id, label, val
FROM mozdata.org_mozilla_firefox.metrics
LEFT JOIN UNNEST(metrics.labeled_counters.some_labeled_counter)
```

Will result in:
```
document_id | label | value
abc | label1 | val1
abc | label2 | val2
abc | label3 | val3
```
So the docid does not uniquely identify a row.

Reply:
I don't think there's actually a primary key anywhere, but the join condition will complain for anything that isn't a cross join.

Reply:
Can you show what you tried without a JOIN key and what the error was?

Reply:
I think I understand this better, it is the primary key for a query like this:

```
with labeled_counter as (
  SELECT document_id, label, val
  FROM mozdata.org_mozilla_firefox.metrics t
  LEFT JOIN UNNEST(metrics.labeled_counters.some_labeled_counter)
),
counter as (
  SELECT document_id, metrics.counter.some_counter
  FROM mozdata.org_mozilla_firefox.metrics t
)
select document_id, counter, label, val
from counter
left join labeled_counter
using (document_id)
```

Which i think is the conceptual join that's happening here, with the symmetric aggregation magic that looker is doing under the hood.